### PR TITLE
WINDUP-498: Application overview report display problems, not solutions

### PR DESCRIPTION
**Pull request by mbriskar**
Show problems, not solutions - Screenshot

What about providing also Location if possible?

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/454/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.tests:tests' /><a href='https://10.3.9.192/job/windup-pr-builder/454/org.jboss.windup.tests$tests/testReport'>org.jboss.windup.tests:tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://10.3.9.192/job/windup-pr-builder/454/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by lincolnthree** [comment]
This breaks a test that needs to be fixed.

**Comment by lincolnthree** [comment]
But nice job! I think we also probably want to use the enhanced JavaTypeReferenceLocationModel description from @jsight 's PR #443 (Please coordinate with him to make that happen.)

**Comment by mbriskar** [comment]
No idea how could this break a test. Isn't that test broken for a longer time?

File: rules-java/src/main/resources/reports/templates/java_application.ftl [hunk 1]
```diff
@@ -47,7 +47,7 @@
           <ul class='notifications'>
             <#list sourceReportModel.sourceFileModel.inlineHints.iterator() as hintLine>
               <#if hintLine.hint?has_content>
-                <li class='warning'>${hintLine.hint}</li>
+                <li class='warning'>${hintLine.fileLocationReference.description}</li>
```
Comment (mbriskar) [inline rules-java/src/main/resources/reports/templates/java_application.ftl]: this waits for PR #443

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/455/<h2>Failed Tests: <span class='status-failure'>12</span></h2><h3><a name='org.jboss.windup.rules.apps:rules-java' /><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.rules.apps$rules-java/testReport'>org.jboss.windup.rules.apps:rules-java</a>: <span class='status-failure'>3</span></h3><ul><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.rules.apps$rules-java/testReport/org.jboss.windup.rules.java/JavaClassXmlRulesTest/testJavaClassCondition/'><strong>org.jboss.windup.rules.java.JavaClassXmlRulesTest.testJavaClassCondition</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.rules.apps$rules-java/testReport/org.jboss.windup.rules.java/JavaHintsClassificationsTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.JavaHintsClassificationsTest.testHintsAndClassificationOperation</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.rules.apps$rules-java/testReport/org.jboss.windup.rules.java.ignore/JavaIgnoreRegexesTest/testHintsAndClassificationOperation/'><strong>org.jboss.windup.rules.java.ignore.JavaIgnoreRegexesTest.testHintsAndClassificationOperation</strong></a></li></ul><h3><a name='org.jboss.windup.tests:tests' /><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport'>org.jboss.windup.tests:tests</a>: <span class='status-failure'>9</span></h3><ul><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureMediumBinaryModeTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupArchitectureMediumBinaryModeTest.testRunWindupMedium</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureScanPackagesTest/testRunWindupScanPackages/'><strong>org.jboss.windup.tests.application.WindupArchitectureScanPackagesTest.testRunWindupScanPackages</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSeamBookingSourceTest/testRunWindupSeamBookingSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSeamBookingSourceTest.testRunWindupSeamBookingSourceMode</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSmallBinaryMode2Test/testRunWindupTiny/'><strong>org.jboss.windup.tests.application.WindupArchitectureSmallBinaryMode2Test.testRunWindupTiny</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSmallBinaryModeTest/testRunWindupTiny/'><strong>org.jboss.windup.tests.application.WindupArchitectureSmallBinaryModeTest.testRunWindupTiny</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSoa5GatewaysRoutersTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSoa5GatewaysRoutersTest.testRunWindupSourceMode</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li><li><a href='https://10.3.9.192/job/windup-pr-builder/455/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

File: config/api/src/main/java/org/jboss/windup/config/phase/Initialization.java [hunk 2]
```diff
@@ -5,7 +5,6 @@
 import org.jboss.windup.config.WindupRuleProvider;
 
 /**
- * Previous: {@link ArchiveExtraction}<br/>
```
Comment (jsight) [inline config/api/src/main/java/org/jboss/windup/config/phase/Initialization.java]: Thanks!

File: rules-java/src/main/java/org/jboss/windup/rules/apps/java/scan/provider/DiscoverFilesAndTypesRuleProvider.java [hunk 3]
```diff
@@ -32,8 +32,7 @@ public Configuration getConfiguration(GraphContext context)
         .when(Query.fromType(FileModel.class)
             .withProperty(FileModel.IS_DIRECTORY, true)
         )
-        .perform(Iteration.over(FileModel.class)
-            .perform(new RecurseDirectoryAndAddFiles()).endIteration()
+        .perform(new RecurseDirectoryAndAddFiles()
```
Comment (jsight) [inline rules-java/src/main/java/org/jboss/windup/rules/apps/java/scan/provider/DiscoverFilesAndTypesRuleProvider.java]: Great! :)

File: rules-java/src/main/resources/reports/templates/java_application.ftl (hunk 1 above)
Comment (jsight) [inline rules-java/src/main/resources/reports/templates/java_application.ftl]: Ok, that's merged. I'll rerun the unit tests and merge now.

**Comment by jsight** [comment]
retest

**Comment by mbriskar** [comment]
tested it locally and it looks the way it should

File: rules-java/src/main/resources/reports/templates/java_application.ftl (hunk 1 above)
Comment (lincolnthree) [inline rules-java/src/main/resources/reports/templates/java_application.ftl]: Nice job, guys :)

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://10.3.9.192//job/windup-pr-builder/456/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.tests:tests' /><a href='https://10.3.9.192/job/windup-pr-builder/456/org.jboss.windup.tests$tests/testReport'>org.jboss.windup.tests:tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://10.3.9.192/job/windup-pr-builder/456/org.jboss.windup.tests$tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by jsight** [comment]
Fixed last unit test and merged.

**Pull request merged**